jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/UserTaskServiceImpl.java

```diff
@@ -192,6 +193,17 @@ public void claim(String deploymentId, Long taskId, String userId) {
 		}
 
 	}
+	
+	@Override
+    public void claim(String deploymentId, Collection<Long> taskIds, String userId) {
+        taskIds.forEach(taskId -> {
+            try {
+                claim(deploymentId, taskId, userId);
+            } catch (TaskNotFoundException e) {
```

Comment:
if the task is not in the expected status to be claimed, it could also throw other exceptions like `PermissionDeniedException`. In that case, it wouldn't continue with the loop.
You could also add a test case for this scenario, by claiming a "Reserved" task.

Reply:
But I think thats right, in that case we can cancel the the operation. @elguardian wdyt?

Reply:
hmmm, I think that since this is launching individual command, we can catch `PermissionDeniedException` and log it. Any other exception you can think of @gmunozfe ?

Reply:
my understanding was that bulk operations are best-effort, trying to complete as many individual operations as possible

Reply:
@fjtirado The other exception I can see is `IllegalStateException`: "Task instance ... is owned by another deployment expected "

Reply:
hmmmm, problem is that illegalstate can hide a different issue, and in any case it will a gross failure and I think we should report